dao/src/main/java/org/thingsboard/server/dao/mobile/BaseMobileAppSettingsService.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright © 2016-2024 The Thingsboard Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.server.dao.mobile;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.event.TransactionalEventListener;
+import org.thingsboard.server.common.data.id.TenantId;
+import org.thingsboard.server.common.data.mobile.AndroidConfig;
+import org.thingsboard.server.common.data.mobile.BadgePosition;
+import org.thingsboard.server.common.data.mobile.BadgeStyle;
+import org.thingsboard.server.common.data.mobile.IosConfig;
+import org.thingsboard.server.common.data.mobile.MobileAppSettings;
+import org.thingsboard.server.common.data.mobile.QRCodeConfig;
+import org.thingsboard.server.dao.entity.AbstractCachedService;
+
+import static org.thingsboard.server.dao.service.Validator.validateId;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class BaseMobileAppSettingsService extends AbstractCachedService<TenantId, MobileAppSettings, MobileAppSettingsEvictEvent> implements MobileAppSettingsService {
+
+    public static final String INCORRECT_TENANT_ID = "Incorrect tenantId ";
+    private static final String DEFAULT_QR_CODE_LABEL = "Scan to connect or download mobile app";
+    private final MobileAppSettingsDao mobileAppSettingsDao;
+
+    @Override
+    public MobileAppSettings saveMobileAppSettings(TenantId tenantId, MobileAppSettings settings) {
+        MobileAppSettings mobileAppSettings = mobileAppSettingsDao.save(tenantId, settings);
```

Comment:
Where is the validation? You marked some fields with `@NoXss`, so you need to call `ConstraintValidator`.